src/io/rainrobot/maxdispachreq/SessionCache.java

```diff
@@ -0,0 +1,35 @@
+package io.rainrobot.maxdispachreq;
+
+import java.util.Map;
+
+public class SessionCache {
+    private Map<String, Session> tokenSessionMap;
+
+
+    public boolean exist(String token) {
+        Session session = tokenSessionMap.get(token);
+        //if exist & not expired
+        if(session != null && !session.isExpired()) {
```

Comment:
not clear how timeout is working, it doesn't seems that you update the alast touched ttime.
when you doing something with expiration but resource allocated to it you usually need to update it was used so, you won't change it while it is still active and break it

Reply:
Do you mean it should update the expiration with each client request? I made it so it has 24h when the session is created and this is fixed.

from AddressService:
```
private SimpleSession buildSession(String token, ServerAddress adders) {
        LocalDateTime experation = LocalDateTime.now().plusHours(24);
        experationSchdualer.setExperation(experation, () -> endSession(token));
        return new SimpleSession.Builder()
                .token(token).addres(adders).experation(experation).build();
    }
```